source/main.civet

```diff
@@ -81,16 +83,53 @@ export compile := (src: string, options?: CompilerOptions) ->
   if filename.endsWith('.coffee') and not /^(#![^\r\n]*(\r\n|\n|\r))?\s*['"]civet/.test src
     options.parseOptions.coffeeCompat = true
 
-  let events
-  if !options.noCache
-    events = makeCache()
+  {hits, trace, noCache} := options
 
-  //@ts-ignore
-  parse.config = options.parseOptions or {}
-  ast := prune parse(src, {
-    filename
-    events
-  })
+  let events: CacheEvents | undefined
+  unless noCache
+    events = makeCache {
+      hits: !!hits
+      trace: !!trace
+    }
+
+  let ast
+
+  try
+    //@ts-ignore
+    parse.config = options.parseOptions or {}
+    ast = prune parse(src, {
+      filename
+      events
+    })
+  catch err
+    throw err
```

Comment:
These two lines aren't needed (and will perhaps overcomplicate stack traces): you can just do `try...finally`.